msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -211,7 +247,71 @@ public void signIn(@NonNull final Activity activity,
                 null // claimsRequest
         );
 
-        acquireTokenInternal(acquireTokenParameters, PublicApiId.SINGLE_ACCOUNT_PCA_SIGN_IN);
+        acquireTokenInternal(acquireTokenParameters, SINGLE_ACCOUNT_PCA_SIGN_IN);
+    }
+
+    @Override
+    public void signIn(@NonNull final Activity activity,
+                       @Nullable final String loginHint,
+                       @NonNull final String[] scopes,
+                       @Nullable final Prompt prompt,
+                       @NonNull final AuthenticationCallback callback) {
+        final IAccount persistedAccount = getPersistedCurrentAccount();
+
+        if (persistedAccount != null) {
+            callback.onError(
+                    new MsalClientException(
+                            MsalClientException.INVALID_PARAMETER,
```

Comment:
Actually, I think INVALID_PARAMETER could be misleading. Let's create an opposite of MsalClientException.NO_CURRENT_ACCOUNT. 

Maybe ALREADY_SIGNED_IN?

Reply:
+1

Reply:
@rpdome @shahzaibj Just so we're clear, you are both suggesting a breaking change. See current implementation of Single Account

Do you really want this change?

Reply:
We can keep it as is